index.rst

```diff
@@ -1,51 +1,1667 @@
 
-  Do not put the title, authors or other metadata in this document;
-  those are automatically added.
+.. _scope:
 
-  Use the following syntax for sections:
+Scope of the document
+=====================
 
-  Sections
-  ========
+This document is designed to assist developers involved in porting LSST Science Pipelines
+from swig to pybind11 by demonstrating how to wrap a package in pybind11.
+It assumes that the reader has already read (but not necessarily completely understood)
+the pybind11 documentation and the LSST coding guidelines (see `additional`_).
 
-  and
+.. _intro:
 
-  Subsections
-  -----------
+.. _additional:
 
-  and
+Additional documentation
+========================
 
-  Subsubsections
-  ^^^^^^^^^^^^^^
+Also see the:
 
-  To add images, add the image file (png, svg or jpeg preferred) to the
-  _static/ directory. The reST syntax for adding the image is
+* `The pybind11 upstream documentation <http://pybind11.readthedocs.io>`_ 
+* `The LSST pybind11 coding guidelines <https://dmtn-024.lsst.io>`_ 
 
-  .. figure:: /_static/filename.ext
-     :name: fig-label
-     :target: http://target.link/url
+.. _installation:
 
-     Caption text.
+Installation
+============
 
-   Run: ``make html`` and ``open _build/html/index.html`` to preview your work.
-   See the README at https://github.com/lsst-sqre/lsst-technote-bootstrap or
-   this repo's README for more info.
+To install all of the currently wrapped code with pybind11, use
 
-   Feel free to delete this instructional comment.
+.. code-block:: bash
 
-:tocdepth: 1
-.. Please do not modify tocdepth; will be fixed when a new Sphinx theme is shipped.
+    rebuild -r tickets/DM-8467 {{package name}}
+
+where ``{{package name}}`` is the name of the package that is currently being wrapped (for instance ``afw``).
+This will build the most up to date version of the stack that has been ported to pybind11, 
+as the tests that have not been wrapped are all commented out (see section :ref:`activate_test`).
+
+Don't forget to tag the new build as current with EUPS:
+
+.. code-block:: bash
+
+    eups tags --clone bNNNN current
+
+where ``bNNN`` is the current build number.
+
+.. _new_package:
+
+Wrapping a new package
+======================
+
+Since many packages have C++ classes and functions that are not exposed to Python, 
+there are large chunks of C++ code that are not currently tested explicitly.
+Without tests there is no way to know whether the wrapping was successfull, 
+so for now we only wrap C++ code that are called from Python tests.
+A single test might import from multiple submodules of the current package, 
+so we found that it is more efficient to wrap by test as opposed to wrapping by module.
+The following section outlines the procedure to begin wrapping a new package.
+
+Preparing Package
+-----------------
+
+Before you can begin wrapping a package it is necessary to modify the structure of the package,
```

Comment:
Good point. I didn't think about this since I started wrapping afw after Pim had already done the initial setup, so I just neglected to add these steps.